File: typescript/src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/elements/Circle.ts
import {Shape} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/elements/Shape";
import {Visitor} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/visitors/Visitor";

/** Concrete Element */
class Circle implements Shape {
    move(x: number, y: number): void {
        // move shape
    }

    draw(): void {
        // draw shape
    }

    accept(visitor: Visitor): void {
        visitor.visitCircle(this);
    }
}

export { Circle };

File: java/src/Basic_Patterns/Behavioural_Patterns/Double_Dispatch/Problem_Showcase/Elements/Circle.java
package Basic_Patterns.Behavioural_Patterns.Double_Dispatch.Problem_Showcase.Elements;

public class Circle extends Dot {
    protected int radius;

    public Circle(String id, int x, int y, int radius) {
        super(id, x, y);
        this.radius = radius;
    }

    @Override
    public void draw() {
        System.out.println("Drawing a circle at (" + x + ", " + y + ") with radius " + radius + ".");
    }
}
